demo/src/faber.ts

```diff
@@ -0,0 +1,168 @@
+import type { ConnectionRecord } from '@aries-framework/core'
+import type { CredDef, Schema } from 'indy-sdk-react-native'
+import type BottomBar from 'inquirer/lib/ui/bottom-bar'
+
+import {
+  ConnectionInvitationMessage,
+  CredentialPreview,
+  JsonTransformer,
+  ProofAttributeInfo,
+  AttributeFilter,
+} from '@aries-framework/core'
+import { uuid } from '@aries-framework/core/build/utils/uuid'
+import { JsonEncoder } from '@aries-framework/core/src/utils/JsonEncoder'
+import { ui } from 'inquirer'
+
+import { BaseAgent } from './base_agent'
+import { Color, Output } from './output_class'
+
+export class Faber extends BaseAgent {
+  public connectionRecordAliceId?: string
+  public credentialDefinition?: CredDef
+  public ui: BottomBar
+
+  public constructor(port: number, name: string) {
+    super(port, name)
+    this.ui = new ui.BottomBar()
+  }
+
+  public static async build(): Promise<Faber> {
+    const faber = new Faber(9001, 'faber')
+    await faber.initializeAgent()
+    return faber
+  }
+
+  private async getConnectionRecord() {
+    if (!this.connectionRecordAliceId) {
+      throw Error(`${Color.red}${Output.missingConnectionRecord}${Color.reset}`)
+    }
+    return await this.agent.connections.getById(this.connectionRecordAliceId)
+  }
+
+  private async receiveConnectionRequest(invitation_url: string) {
+    const http = 'http://localhost:9000?c_i='
+    let invitationJson = invitation_url.replace(http, '')
+
+    try {
+      invitationJson = JsonEncoder.fromBase64(invitationJson)
+    } catch (e) {
+      console.log(`${Color.green}\nIt looks like your invitation link is not correctly formatted?\n${Color.reset}`)
+      return
+    }
+
+    const invitationMessage = JsonTransformer.fromJSON(invitationJson, ConnectionInvitationMessage)
+    return await this.agent.connections.receiveInvitation(invitationMessage)
```

Comment:
I think we can simplify this with the `receiveInvitationFromUrl` method

```suggestion
    return await this.agent.connections.receiveInvitationFromUrl(invitationUrl)
```